src/lib/languages/javascript.rs

```diff
@@ -0,0 +1,144 @@
+use super::*;
+
+pub struct JavaScript {
+  pub src: String
+}
+
+impl JavaScript {
+  // Generate javascript code using tokens from parser
+  pub fn generate(parser: Parser) -> Result<Self, ParsingError> {
+    let mut code = Self{
+      src: String::new()
+    };
+    // define functions
+    for func in parser.functions {
+      let function = code.function(func);
+      code.src += &function;
+    }
+    for glob in parser.global_vars {
+      let global = code.global(glob);
+      code.src += &global;
+    }
+    // Because JS does not call main, we must do that here
+    code.src += "main();\n";
+    Ok(code)
+  }
+  pub fn function(&mut self, func: Function) -> String {
+    let mut src = String::new();
+    src += "function ";
+    src += &func.name.unwrap();
+    src += "(";
+    let mut index = 0;
+    while index < func.args.len() {
+      let arg = &func.args[index];
+      src += &arg.0;
+      src += ",";
+      index += 1;
+    }
+    src.pop();
+    src += ") {\n";
+    let actions = func.body.list;
+    for action in actions {
+      src += &self.action(action);
+    }
+    src += "}\n";
+    return src;
+  }
+  pub fn global(&mut self, var: Variable) -> String {
+    let mut src = String::new();
+    match var.var_type {
+      variable::VarType::Const => {src += "const ";}
+      _ => {/* Add error */}
+    }
+    src += &var.name.to_string();
+    src += " = ";
+    src += &self.action(*var.action);
+    src += ";\n";
+    return src;
+  }
+  pub fn action(&mut self, action: Action) -> String {
+    // match an action and return code
+    let mut src = String::new();
+    match action {
+      Action::Break => src += "break;\n",
+      Action::Continue => src += "continue;\n",
+      Action::For(res) => src += &self.action_for(res),
+      Action::FunctionCall(res) => src += &self.action_func_call(res),
+      Action::Loop(res) => src += &self.action_loop(res),
+      Action::Return(res) => src += &self.action_return(res),
+      Action::StaticNumber(res) => src += &self.action_num(res),
+      Action::StaticString(res) => src += &self.action_str(res),
+      Action::Variable(res) => src += &self.action_var(res),
+      Action::VarRef(res) => src += &res,
+      // add remaining actions
+      _ => {src += "undefined\n"}
```

Comment:
Can you not add this match all statement here?  
I rather prefer the code under here because if we then later add another action it's supper easy to see where you need to update the code to support the new action.  
```rust
    Action::Name1(_) | Action::Name2(_) | Action::Name3 (_) => 
```